### PR TITLE
feat(connect): add descriptor for taproot xpub

### PR DESCRIPTION
**Pull request by peter-sanderson**
Adds a descriptor for taproot xpub based on https://github.com/trezor/trezor-firmware/pull/3545

Resolves: https://github.com/trezor/trezor-suite/issues/10963

image

### CODE REVIEW

File: packages/suite/src/components/suite/modals/ReduxModal/ConfirmXpubModal.tsx [hunk 1]
```diff
@@ -33,7 +33,11 @@ export const ConfirmXpubModal = (
             confirmStepLabel={<Translation id="TR_XPUB_MATCH" />}
             validateOnDevice={showXpub}
             copyButtonText={<Translation id="TR_XPUB_MODAL_CLIPBOARD" />}
-            value={account.descriptor}
+            value={
+                account.descriptorChecksum !== undefined
+                    ? `${account.descriptor}#${account.descriptorChecksum}`
+                    : account.descriptor
+            }
```
Comment (komret) [inline packages/suite/src/components/suite/modals/ReduxModal/ConfirmXpubModal.tsx]: nit: I'd do this in the component body.

File: packages/connect/src/device/resolveDescriptorForTaproot.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        };
+    } else {
+        // wrap regular xpub into bitcoind native descriptor
+        const fingerprint = Number(publicKey.root_fingerprint || 0)
+            .toString(16)
+            .padStart(8, '0');
+        const descriptorPath = `${fingerprint}${response.serializedPath.substring(1)}`;
+
+        return {
+            xpubSegwit: `tr([${descriptorPath}]${response.xpub}/<0;1>/*)`,
```
Comment (komret) [inline packages/connect/src/device/resolveDescriptorForTaproot.ts]: 🤔 What makes this a Segwit XPUP? I don't get the naming here.

**Review by komret — CHANGES_REQUESTED** [review]
Screenshot 2024-03-01 at 15 23 39

It doesn't work for me, using the latest emulator. The checksum is displayed on the device but not in Suite.

**Comment by komret** [comment]
The base branch should be `develop`:

Screenshot 2024-03-01 at 15 30 14

File: packages/connect/src/device/resolveDescriptorForTaproot.ts (hunk 2 above)
Comment (peter-sanderson) [reply packages/connect/src/device/resolveDescriptorForTaproot.ts]: This is the original implementation that I kept to maintain backward compatibility.

Comment (peter-sanderson) [reply packages/connect/src/device/resolveDescriptorForTaproot.ts]: But, true - I renamed it to just `xpub`

**Review by komret — APPROVED** [review]
Emulator doesn't have the latest FW from `main` because their pipelines have been red for weeks - needs to be tested with FW 2.6.5.

**Pull request merged**